libs/components/src/lib/file-picker/file-picker.ts

```diff
@@ -255,10 +241,19 @@ export class FilePicker extends FormAssociatedFilePicker {
 			this.setFormValue(formData);
 		}
 
-		// For historical reasons, the value IDL attribute prefixes the filename with the string "C:\fakepath\".
+		// Like native input[type=file], set the value to a fakepath
 		this.value = files.length > 0 ? `C:\\fakepath\\${files[0].name}` : '';
 	}
 
+	override setFormValue = (value: File | string | FormData | null, state?: File | string | FormData | null) => {
+		if (typeof value === 'string') {
+			// Ignore FormAssociated trying to set the form value to the fakepath
+			return;
+		}
+
+		super.setFormValue(value, state);
```

Comment:
Extract logic to global module function (or a private one on the class). 
Self-descriptive and no need to read it unless one really needs to.

```suggestion
		if (formAssociationInjnectsFakePath(value)) {
			return;
		}

		super.setFormValue(value, state);
```